File: backend/app/models/channels/channel.py
# Copyright (c) 2026 Valaris Studio
# SPDX-License-Identifier: AGPL-3.0-or-later

import enum
import uuid

from sqlalchemy import Enum, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.types import JSON

from app.models.base import Base, TimestampMixin, UUIDMixin


class ChannelType(str, enum.Enum):
    email = "email"
    slack = "slack"
    whatsapp = "whatsapp"
    phone = "phone"
    website = "website"
    other = "other"


class Channel(Base, UUIDMixin, TimestampMixin):
    __tablename__ = "channels"

    workspace_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("workspaces.id", ondelete="CASCADE"), index=True
    )
    name: Mapped[str] = mapped_column(String(255))
    channel_type: Mapped[ChannelType] = mapped_column(Enum(ChannelType))
    contact_value: Mapped[str] = mapped_column(String(500))
    description: Mapped[str] = mapped_column(Text, default="")
    metadata_json: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    created_by: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id")
    )

File: frontend/src/types/channel.ts
// Copyright (c) 2026 Valaris Studio
// SPDX-License-Identifier: AGPL-3.0-or-later

export type ChannelType = "email" | "slack" | "whatsapp" | "phone" | "website" | "other";

export interface Channel {
  id: string;
  workspace_id: string;
  name: string;
  channel_type: ChannelType;
  contact_value: string;
  description: string;
  metadata_json: Record<string, unknown> | null;
  created_by: string;
  created_at: string;
  updated_at: string;
}
